solar_system.rb

```diff
@@ -0,0 +1,41 @@
+require 'minitest'
+
+class SolarSystem
+  attr_reader :star_name, :planets
+
+  def initialize(star_name)
+    @star_name = star_name
+    @planets = []
+  end
+
+  def add_planet(planet)
+    @planets << planet
+  end
+
+  def list_planets
+    list = "Planets orbiting #{@star_name}: \n"
+    @planets.each_with_index do |planet, index|
+      list += "#{index + 1}. #{planet.name} \n"
+    end
+    return list
+  end
+
+  def find_planet_by_name(planet_name)
+    planet_name.downcase!
+    planet_name.capitalize!
+
+    planet = @planets.select { |planet| planet.name == planet_name }
```

Comment:
You could also use `.find` which returns the first one that matches and `nil` if nothing matches.